synapse_antispam/mjolnir/antispam.py

```diff
@@ -115,6 +115,11 @@ def user_may_create_room_alias(self, user_id, room_alias):
     def user_may_publish_room(self, user_id, room_id):
         return True  # allowed
 
+    def check_username_for_spam(self, user_id, display_name):
+        # Check whether the user ID or display name matches any of the banned
+        # patterns.
+        return self.is_user_banned(user_id) or self.is_user_banned(display_name)
```

Comment:
We should put this check behind a config flag (just in case someone doesn't want it for some reason). Similar to the `block_invites` config flag already in place.